control_panel_api/serializers.py

```diff
@@ -266,3 +269,41 @@ class Meta:
             'nickname',
             'name',
         )
+
+
+class ESBucketHitsSerializer(serializers.BaseSerializer):
+
+    def to_representation(self, aggregations):
+        accessed_by_sum_map = {}
+        accessed_by_type_map = {}
+
+        for result in aggregations.bucket_hits:
+            role_type, accessed_by = self._extract_fields(result.key)
+
+            accessed_by_type_map[accessed_by] = role_type
+
+            if accessed_by not in accessed_by_sum_map:
+                accessed_by_sum_map[accessed_by] = result.doc_count
+            else:
+                accessed_by_sum_map[accessed_by] += result.doc_count
+
+        results = [
+            {'accessed_by': k, 'count': v, 'type': accessed_by_type_map[k]}
+            for k, v in accessed_by_sum_map.items()
+        ]
+
+        sorted_by_sum = sorted(results, key=itemgetter('count'), reverse=True)
+
+        return sorted_by_sum
+
+    def _extract_fields(self, key):
```

Comment:
This function name is a bit generic and not intent-revealing. Maybe something like `get_accessed_by`?

Reply:
Or maybe something like `get_role_name_and_type()`...not sure about the name but I would definitely keep the `_` prefix so it's clear it's only used internally and the future us know they can go nuts on refactoring/renaming if necessary.